apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/ClusterService.java

```diff
@@ -26,7 +26,7 @@ public ClusterService(final UserInfoHolder userInfoHolder, final AdminServiceAPI
     return clusterAPI.findClustersByApp(appId, env);
   }
 
-  public ClusterDTO createCluster(Env env, ClusterDTO cluster) {
+  public synchronized ClusterDTO createCluster(Env env, ClusterDTO cluster) {
```

Comment:
I don't think adding `synchronized` here will solve the concurrency problem because there may be more than 1 apollo-portal instances running. That should be solved in apollo-adminservice, probably with a distributed lock.

Reply:
Emmm. It seems like that. I will change the code back.